redfish/softwareinventory.go

```diff
@@ -0,0 +1,62 @@
+//
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+package redfish
+
+import (
+	"encoding/json"
+
+	"github.com/stmcginnis/gofish/common"
+)
+
+// SoftwareInventory is used to represent a software or firmware item on the server
+type SoftwareInventory struct {
+	common.Entity
+
+	Description            string
+	Status                 common.Status
+	Version                string
+	LowestSupportedVersion string
+	Manufacturer           string
+	ReleaseDate            string
+	SoftwareID             string
+	UefiDevicePaths        []string
+	Updateable             bool
+	WriteProtected         bool
+	rawData                []byte
+}
+
+// UnmarshalJSON unmarshals a SoftwareInventory object from the raw JSON
+func (softwareInventory *SoftwareInventory) UnmarshalJSON(b []byte) error {
+	type temp SoftwareInventory
+	var t struct {
+		temp
+	}
+
+	err := json.Unmarshal(b, &t)
+	if err != nil {
+		return err
+	}
+
+	// Extract the links to other entities for later
+	*softwareInventory = SoftwareInventory(t.temp)
+	softwareInventory.rawData = b
```

Comment:
Then if we don't need to capture rawData, the whole UnmarshallJSON call can be removed too and just go with the default unmarshalling.